driver/src/test/java/org/neo4j/driver/v1/GraphDatabaseTest.java

```diff
@@ -156,4 +158,28 @@ public void shouldLogWhenUnableToCreateRoutingDriver() throws Exception
         assertEquals( 0, server1.exitStatus() );
         assertEquals( 0, server2.exitStatus() );
     }
+
+    @Test
+    public void shouldRespondToInterruptsWhenConnectingToUnresponsiveServer() throws Exception
+    {
+        try ( ServerSocket serverSocket = new ServerSocket( 0 ) )
+        {
+            // setup other thread to interrupt current thread when it blocks
+            TestUtil.interruptWhenInWaitingState( Thread.currentThread() );
+            try
+            {
+                GraphDatabase.driver( "bolt://localhost:" + serverSocket.getLocalPort() );
+                fail( "Exception expected" );
+            }
+            catch ( Exception ignore )
+            {
+                // expected
```

Comment:
@lutovich Would it be better to expect a `ServiceUnavailableException` here instead of a `IllegalStateException`?

Reply:
Yeah, that's true. What do you think about always wrapping exceptions from connectivity verification in `ServiceUnavailableException`? (unless it's already a `ServiceUnavailableException`)